pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -60,11 +70,37 @@ def on_trainer_init(
         self.trainer.reload_dataloaders_every_n_epochs = reload_dataloaders_every_n_epochs
         self.trainer._is_data_prepared = False
 
-    def get_profiled_train_dataloader(self, train_dataloader) -> Iterable:
-        self.data_fetcher = DataFetcher()
-        self.data_fetcher.setup(train_dataloader)
+    def _check_training_step_requires_dataloader_iter(self) -> bool:
+        self.trainer.lightning_module
+        training_step_fx = getattr(self.trainer.lightning_module, "training_step")
+        contains_datalaoder_iter = is_param_in_hook_signature(training_step_fx, "dataloader_iter", explicit=True)
+        use_manual_optimization = not self.trainer.lightning_module.automatic_optimization
+        return contains_datalaoder_iter and use_manual_optimization and self.trainer.training
+
+    def _select_data_fetcher(self) -> AbstractDataFetcher:
+        if self._check_training_step_requires_dataloader_iter():
+            log.warning(
+                "Found `dataloader_iter` argument in the `training_step`. Note that the support for "
+                "this signature is experimental and the behavior may subject to change."
+            )
+            return DataLoaderIterDataFetcher()
+        elif os.getenv("PL_INTER_BATCH_PARALLELISM", "0") == "1":
+            # FIXME: Replace by self.trainer.training_type_plugin.on_gpu
```

Comment:
Just to confirm what this means, this means later to check if we're on GPU as well as inter batch parallelism has been enabled?

Reply:
Replace by:
```py
self.trainer.training_type_plugin.on_gpu and os.getenv("PL_INTER_BATCH_PARALLELISM", "0") == "1":
```